### ISSUE TITLE
Width/Height attributes only set on first save & off by one

### ISSUE THREAD

**Issue by humdingerb** [HaikuArchives/ArtPaint#424]
* Create a 640x360 canvas
* Save as test.png
* Quit ArtPaint
* In Tracker, with GetInfo the attributes show 639x359
* Open with ShowImage saves the correct attibutes 640x360
* Open test.png in ArtPaint
* Crop to 100x100
* Save image
* In Tracker, with GetInfo the attributes still show 640x360

**Comment by humdingerb** [comment]
As it turns out, this isn't ArtPaint's fault. How can it, Artpaint never writes those attributes...
It seems to be Tracker's new thumbnailer. I'll file a ticket at Trac.

**Comment by dsizzle** [comment]
I saw the code review… I wonder if your fix should be ceil(Width()) to round up any fractional Width values instead of IntegerWidth() + 1 which will always add 1?  (Same for height of course)

**Comment by humdingerb** [comment]
Doesn't the use of IntegerWidth() already get rid of any fractions?

**Comment by dsizzle** [comment]
It does but I don’t know if it’s rounded up or down. I guess I can look at the source code…

**Comment by dsizzle** [comment]
Looks like it doesn’t handle rounding, it just returns the integer version.  This just returns the part before the decimal point (I.e. a Width() of 10.999999 will return an IntegerWidth() of 10)

**Comment by dsizzle** [comment]
See haiku/headers/os/interface/Size.h

**Comment by dsizzle** [comment]
Actually maybe a better fix overall would be to change the IntegerWidth and IntegerHeight functions to add 0.5 to the width and height which would then go the rounding 

so like: 

`return (int32)(width + 0.5);`

since width and height should only be > 0 (this doesn’t work for negative numbers)

**Comment by dsizzle** [comment]
I see waddlesplash disagrees with me - but it is good to know that width is pixel center-to-pixel center...so two pixels wide = width of 1.  Might explain some of the weird off-by-ones in ArtPaint!

**Comment by humdingerb** [comment]
Yes, that center-to center has been a fun source of issues since BeOS days. :)
Might be the reason for 1 pixel selections not working...
